In [4]:
import pandas as pd
import numpy as np
import os

In [6]:
path=r'C:\Users\aditi\Desktop\Data Analytics_Career Foundry\Data Analytics_Immersion\Project 4\Instacart Basket Analysis'

In [8]:
# Task 3

df_ords_prods_combined=pd.read_pickle(os.path.join(path, 'Data', 'Prepared data', 'orders_products_combined.pkl'))

In [10]:
# Task 4
df_ords_prods_combined.shape

(32434489, 12)

In [12]:
df_prods=pd.read_csv(os.path.join(path, 'Data', 'Prepared data', 'products_checked.csv'), index_col = False)

In [14]:
df_prods.shape

(49672, 6)

In [16]:
df_ords_prods_combined.head()

Unnamed: 0.1  Unnamed: 0  order_id  user_id  order_number  \
0             0           0   2539329        1             1   
1             0           0   2539329        1             1   
2             0           0   2539329        1             1   
3             0           0   2539329        1             1   
4             0           0   2539329        1             1   

   orders_day_of_week  order_hour_of_day  days_since_prior_order  product_id  \
0                   2                  8                     0.0         196   
1                   2                  8                     0.0       14084   
2                   2                  8                     0.0       12427   
3                   2                  8                     0.0       26088   
4                   2                  8                     0.0       26405   

   add_to_cart_order  reordered _merge  
0                  1          0   both  
1                  2          0   both  
2                  3          0   both  
3                  4          0   both  
4                  5          0   both

In [18]:
df_prods.head()

Unnamed: 0  product_id                                       product_name  \
0           0           1                         Chocolate Sandwich Cookies   
1           1           2                                   All-Seasons Salt   
2           2           3               Robust Golden Unsweetened Oolong Tea   
3           3           4  Smart Ones Classic Favorites Mini Rigatoni Wit...   
4           4           5                          Green Chile Anytime Sauce   

   aisle_id  department_id  prices  
0        61             19     5.8  
1       104             13     9.3  
2        94              7     4.5  
3        38              1    10.5  
4         5             13     4.3

In [41]:
# Task 5
df_drop_merge=df_ords_prods_combined.drop('_merge',axis=1)

In [43]:
# Task 5
df_prods_ords_merge=df_prods.merge(df_drop_merge, on = 'product_id', indicator = True)

In [24]:
df_prods_ords_merge.head()

Unnamed: 0_x  product_id                product_name  aisle_id  \
0             0           1  Chocolate Sandwich Cookies        61   
1             0           1  Chocolate Sandwich Cookies        61   
2             0           1  Chocolate Sandwich Cookies        61   
3             0           1  Chocolate Sandwich Cookies        61   
4             0           1  Chocolate Sandwich Cookies        61   

   department_id  prices  Unnamed: 0.1  Unnamed: 0_y  order_id  user_id  \
0             19     5.8          1987          1987   3139998      138   
1             19     5.8          1989          1989   1977647      138   
2             19     5.8         11433         11433    389851      709   
3             19     5.8         12198         12198    652770      764   
4             19     5.8         12200         12200   1813452      764   

   order_number  orders_day_of_week  order_hour_of_day  \
0            28                   6                 11   
1            30                   6                 17   
2             2                   0                 21   
3             1                   3                 13   
4             3                   4                 17   

   days_since_prior_order  add_to_cart_order  reordered _merge  
0                     3.0                  5          0   both  
1                    20.0                  1          1   both  
2                     6.0                 20          0   both  
3                     0.0                 10          0   both  
4                     9.0                 11          1   both

In [45]:
# Task 6
df_prods_ords_merge['_merge'].value_counts()

_merge
both          32404859
left_only            0
right_only           0
Name: count, dtype: int64

In [49]:
# Task 6
test_merge=pd.merge(df_prods, df_drop_merge, on='product_id', how='outer', indicator=True)

In [37]:
test_merge['_merge'].value_counts()

_merge
both          32404859
right_only       30200
left_only           11
Name: count, dtype: int64

The above command shows that there are 32404859 rows common in both 'products' and 'order_products_combined' datasets. Apart from that, there are 30200 rows in df_drop_merge and 11 rows in df_prods, that did not match. For this exercise, we are concerned only about inner join (where both data sets matched).

In [51]:
# Exporting merged file

df_prods_ords_merge.to_pickle(os.path.join(path, 'Data','Prepared data', 'ords_prods_merge.pkl'))